### DISCUSSION TITLE
@nx/vite:build: `{workspaceRoot}` is not honored in `outputPath`

### DISCUSSION THREAD

**Discussion by jsmecham** [nrwl/nx#28661] [Feature Requests]
### Current Behavior

When configuring a build target with the `@nx/vite:build` with `outputPath` set to `{workspaceRoot}/dist`, the actual output occurs in `{projectRoot}/dist`.

### Expected Behavior

I would expect that `outputPath` would respect `{workspaceRoot}`.

### Steps to Reproduce

1.

### Nx Report

Node           : 20.18.0
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 10.9.0

nx                 : 20.0.1
@nx/js             : 20.0.1
@nx/eslint         : 20.0.1
@nx/workspace      : 20.0.1
@nx/cypress        : 20.0.1
@nx/devkit         : 20.0.1
@nx/eslint-plugin  : 20.0.1
@nx/storybook      : 20.0.1
@nx/vite           : 20.0.1
typescript         : 5.6.3
---------------------------------------
Community plugins:
nx-stylelint : 18.0.0

### Package Manager Version

10.9.0

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by Coly010** [comment]
Perhaps confusing in a manner, but `{workspaceRoot}` and `{projectRoot}` are only available for `inputs` and `outputs` to the Nx Task itself. This is used to tell Nx what files will affect the cache hash (inputs) and where the outputted files will be that need to be cached (outputs).

The `outputPath` is an option for the underlying `executor` which knows nothing about Nx caching, nor does it care about it. It takes the options passed to it to configure the build operation. 

Therefore `{workspaceRoot}` nor `{projectRoot}` get interpolated at this point. 
This has always been the case with Nx and therefore, this is not a bug. 

Using Nx Crystal + the `@nx/vite/plugin` inferred task usage (https://nx.dev/concepts/inferred-tasks#inferred-tasks-project-crystal) helps to eliminate this confusion.

I'm going to convert this to a discussion as a feature request, however, I don't think this will get changed. It would mean updating every executor we have to support this new style of defining output paths.
